lib/rules/prefer-arrow-callback.js

```diff
@@ -115,6 +115,17 @@ function getCallbackInfo(node) {
     throw new Error("unreachable");
 }
 
+/**
+* Checks whether a simple list of parameters contains any duplicates. This does not handle complex
+parameter lists (e.g. with destructuring), since complex parameter lists are a SyntaxError with duplicate
+parameter names anyway. Instead, it always returns `false` for complex parameter lists.
+* @param {ASTNode[]} paramsList The list of parameters for a function
+* @returns {boolean} `true` if the list of parameters contains any duplicates
+*/
+function hasDuplicateParams(paramsList) {
```

Comment:
I think this implementation is no problem.
Acorn should be fixed about the duplication error of non-simple parameters.